### PR TITLE
Debug: Improved display params diagnostics

### PR DESCRIPTION
**Pull request by tiran**
The `display_params` decorator now adds two hidden diagnostic flags `--debug-params` and `--debug-params-json` to the list of options. When a flag is passed to a subcommand, it displays all parameters with additional options and then exits with error code 0. The JSON output is designed to be used in tests.

The function now displays more information, too. Besides name and value, it reveals the type of the value and the origin (command line, env vars, default, default, map).

```console
$ ilab model chat --debug-params
Parameters:
  question: ()         [type: tuple, src: commandline]
     model: 'models/merlinite-7b-lab-Q4_K_M.gguf'      [type: str, src: default_map]
   context: 'default'  [type: str, src: default_map]
   session: None       [type: None, src: default]
   ...
```

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [x] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Review by leseb — COMMENTED** [review]
Do we have a troubleshooting guide/doc where these two hidden commands could be documented?

**Comment by tiran** [comment]
No, but we new have a ticket to create one!

File: src/instructlab/utils.py [hunk 1]
```diff
@@ -530,27 +532,96 @@ def http_client(params):
...
+            and logger.isEnabledFor(logging.DEBUG)
+            and not kwargs.get("quiet", False)
+        ):
+            display = "debug"
+
+        if display in {"debug", "human"}:
+            pad = min(max(len(name) for name in kwargs), 24)
+            print("Parameters:")
+            for name, value, typ, src in _get_param_info(ctx, **kwargs):
+                print(f"  {name:>{pad}}: {value!r:<8} \t[type: {typ} src: {src}]")
```
Comment (leseb) [inline src/instructlab/utils.py:595]:
```suggestion
                print(f"  {name:>{pad}}: {value!r:<8} \t[type: {typ} | src: {src}]")
```

Comment (leseb) [reply src/instructlab/utils.py:595]: I'm wondering if this would improve the readability.

File: src/instructlab/utils.py [hunk 2]
```diff
@@ -530,27 +532,96 @@ def http_client(params):
...
+            json.dump({"params": params}, sys.stdout)
+
+        # --display-params* exit
+        if display in {"json", "human"}:
+            ctx.exit()
+
+        return f(*args, **kwargs)
+
+    human_option = click.option(
+        "--display-params",
```
Comment (leseb) [inline src/instructlab/utils.py:610]: Given they are hidden flag intented for debugging I'm wondering it we shouldn't prefix them all with `--debug-` to make it more obvious. WDYT?

**Review by leseb — COMMENTED** [review]
Just nits, it's a great improvement, again! Thanks @tiran

File: src/instructlab/utils.py (hunk 1 above)
Comment (tiran) [reply src/instructlab/utils.py:595]: I'd rather use a comma here. WDYT?

Comment (leseb) [reply src/instructlab/utils.py]: Yes, I'm ok with any separator.

**Review by leseb — DISMISSED** [review]

**Review by RobotSail — DISMISSED** [review]
LGTM

**Review by leseb — APPROVED** [review]

**Review by nathan-weinberg — APPROVED** [review]

**Pull request merged**